Add 65k GKE benchmark

besher-massri opened the pull request:
This PR adds a benchmark for GKE on 65,000 nodes scale

This is following up on the recent [GKE announcement](https://cloud.google.com/blog/products/containers-kubernetes/gke-65k-nodes-and-counting) for support clusters with 65,000-nodes scale.

It benchmarks GKE on a (simulated) AI workload using using Terraform and ClusterLoader2.

I didn't include it as part of `benchmarks` directory as the benchmarking process here is different (using CL2) and has a different infra setup.

andrewsykim commented:
Agree with @achandrasekar that it would be good to put this under the existing `benchmarks/` directory. Is it possible to make that change?

andrewsykim commented on 65k-benchmarks/simulated-ai-workload/README.md:
```diff
@@ -0,0 +1,139 @@
+
+## Results
+
+The benchmark results are stored in the `./output/` directory of the `perf-tests` repository. You can use these results to analyze the performance and scalability of your GKE cluster.
```
Do we plan to publish example results from one of the runs?

besher-massri replied:
Not here, but part of the blogpost. The idea is to link to the blogpost once published.

besher-massri replied:
Is it okay to resolve this ?

besher-massri commented:
> Agree with Ashok Chandrasekar that it would be good to put this under the existing `benchmarks/` directory. Is it possible to make that change?
> 
> > I didn't include it as part of benchmarks directory as the benchmarking process here is different (using CL2) and has a different infra setup.
> 
> I think tooling being different is not a blocker, but would be good to consoildiate eventually

I had an attempt at reorganizing the benchmark, here is a summary of the changes:
- put existing infra under infra/gpu-cluster
- moved infra part under infra, put under folder 65k-cpu-cluster
- moved CL2 part under benchmark/tools
- renamed current README to gpu-benchmark.md
- moved 65k README and renamed into 65k-nodes-simulated-ai-workload
- added a root level README for `benchmarks/` directory

achandrasekar commented:
Thanks for refactoring @besher-massri! @annapendleton can you take a look to make sure this doesn't affect the existing benchmarks?

annapendleton reviewed (changes requested):
This is awesome work! Everything lgtm and shouldn't break existing functionality since it's just migrating locations. Only thing is we want to accurately represent what the infra does so people understand that it supports both GPU and TPU infra

besher-massri commented on benchmarks/gpu-benchmark.md:
````diff
@@ -0,0 +1,231 @@
+
+```sh
+# Stage 2 configures the cluster for running inference workloads.
+cd infra/gpu-cluster/stage-2
````
I went with using accelerator keyword instead, i.e. `accelerator-cluster` as opposed to the other `cpu-cluster`

annapendleton replied:
Thanks, that lgtm!

besher-massri commented on benchmarks/README.md:
````diff
@@ -1,230 +1,10 @@
-terraform destroy
-```
+There are currently the following types of benchmarks available:
+- **GPU-based benchmark**: running inference servers on GPU based clusters. Can be found in `gpu-benchmark.md`
````
I pushed another commit, changing to accelerator and highlighting that both GPU and TPU clusters are supported. Changed some naming, minor rewording as well

besher-massri commented:
Thanks for the review and comments! I pushed another commit, changing from gpu to accelerator and highlighting that both GPU and TPU clusters are supported. Changed some naming, minor rewording as well. Let me know if it's okay.

annapendleton commented:
That looks great, thanks! Please make sure to resolve anything lingering with Andrew prior to submission :)

andrewsykim reviewed (approved):
LGTM, deferring merge to @genlu2011

andrewsykim commented:
/gcbrun
